lib/blocktree/blocktree.go

```diff
@@ -350,7 +370,8 @@ func (bt *BlockTree) DeepCopy() *BlockTree {
 	defer bt.RUnlock()
 
 	btCopy := &BlockTree{
-		db: bt.db,
+		db:         bt.db,
+		blockCache: bt.blockCache,
```

Comment:
this doesn't perform a deep copy, since the nodes in the map still reference the nodes in `bt`. I think you can just remove this for now

Reply:
Oops, i missed that.

Reply:
I added the implementation of the nodeCache copy!